include/fuzzuf/executor/traits/afl_capable.hpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * fuzzuf
+ * Copyright (C) 2022 Ricerca Security
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+/**
+ * @file afl_capable.hpp
+ * @author Ricerca Security <fuzzuf-dev@ricsec.co.jp>
+ */
+
+#ifndef FUZZUF_INCLUDE_EXECUTOR_TRAITS_AFL_CAPABLE_HPP
+#define FUZZUF_INCLUDE_EXECUTOR_TRAITS_AFL_CAPABLE_HPP
+
+#include "fuzzuf/executor/native_linux_executor.hpp"
+#include "fuzzuf/feedback/exit_status_feedback.hpp"
+#include "fuzzuf/feedback/inplace_memory_feedback.hpp"
+#include "fuzzuf/utils/common.hpp"
+
+/**
+ * @class DynAFLCapable
+ * @brief Represents minimal requirements for an AFL-capable executor.
+ * 
+ * @details The executor for AFL must have the methods declared in this class.
+ * This class is to perform type erasure for the executor class to abstract
+ * the executor on the algortihms. C++20 introduces concept that can describe
+ * such type requirements, but fuzzuf uses C++17 by default, so we cannot use
+ * concept at this time. We also found that the boost's type_erasure does not
+ * meet our needs, so we implemented our own type erasure.
+ */
+class DynAFLCapable
+{
+public:
+  template<class T> DynAFLCapable(const T& executor)
+  : _container(new DynContainerDerived<T>(executor))
+  {}
+  ~DynAFLCapable() { delete _container; }
```

Comment:
Since this class is copyable, this causes double free.

Reply:
Thanks, I added some deletion of implicit constructors.

Reply:
Using `std::unique_ptr` instead of raw pointer is safer and clearer in the first place? (is there any problem to do that?

Reply:
Fixed to use `std::unique_ptr` in [6532c86](https://github.com/fuzzuf/fuzzuf/pull/41/commits/6532c867cab29106390551e6f62aa905652fbbfd) and [16bb5ea](https://github.com/fuzzuf/fuzzuf/pull/41/commits/16bb5eaadfce9f423d019f2c9a6187b9ac375998).